### PR TITLE
Address all clippy lints

### PR DESCRIPTION
**Pull request by christianpoveda**

### CODE REVIEW

File: bee-protocol/src/lib.rs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 // Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 // an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and limitations under the License.
```
Comment (thibault-martinez) [inline bee-protocol/src/lib.rs:10]: New line pls :p

**Comment by thibault-martinez** [comment]
Looks like there are 3 clippy errors remaining

File: bee-protocol/src/worker/bundle_validator.rs [hunk 2]
```diff
@@ -31,7 +31,8 @@ impl<N: Node> Worker<N> for BundleValidatorWorker {
     type Config = ();
     type Error = WorkerError;
 
-    async fn start(node: &N, _config: Self::Config) -> Result<Self, Self::Error> {
+    #[allow(clippy::unit_arg)]
```
Comment (thibault-martinez) [inline bee-protocol/src/worker/bundle_validator.rs:34]: Why is this allow here and not in the other workers ?

Comment (thibault-martinez) [reply bee-protocol/src/worker/bundle_validator.rs:34]: Ah you added it crate level, maybe remove this one then ?

File: bee-snapshot/src/global/file.rs [hunk 3]
```diff
@@ -46,12 +47,12 @@ impl GlobalSnapshot {
...
+            let tokens = line.split(';').collect::<Vec<&str>>();
 
-            if tokens.len() < 2 {
-                return Err(Error::MissingSemicolon);
-            } else if tokens.len() > 2 {
-                return Err(Error::ExtraneousSemicolon);
+            match tokens.len().cmp(&2) {
+                Ordering::Less => return Err(Error::MissingSemicolon),
+                Ordering::Greater => return Err(Error::ExtraneousSemicolon),
+                Ordering::Equal => (),
```
Comment (thibault-martinez) [inline bee-snapshot/src/global/file.rs:55]: Hmmm not sure about this one, less readable IMO

Comment (thibault-martinez) [reply bee-snapshot/src/global/file.rs:55]: Please don't be mad but I prefer the previous version :(

Comment (christianpoveda) [reply bee-snapshot/src/global/file.rs:55]: I'm so mad I'm going to change it back to the previous version :trollface:

**Pull request merged**